diff -Naur git-src/platform/qt/src/mbgl/timer.cpp git-src-mod/platform/qt/src/mbgl/timer.cpp
--- git-src/platform/qt/src/mbgl/timer.cpp	2025-01-27 10:02:39.141361000 +0100
+++ git-src-mod/platform/qt/src/mbgl/timer.cpp	2025-01-30 15:18:01.466516800 +0100
@@ -4,6 +4,7 @@
 #include <mbgl/util/timer.hpp>
 
 #include <memory>
+#include <algorithm>
 
 namespace mbgl {
 namespace util {
@@ -19,6 +20,7 @@
 
     timer.setSingleShot(true);
 #if QT_VERSION >= QT_VERSION_CHECK(5, 8, 0)
+    timeout = std::min(timeout, uint64_t(std::numeric_limits<int>::max())); // avoid int overflows causing negative timers
     timer.start(static_cast<std::chrono::milliseconds>(timeout));
 #else
     timer.start(static_cast<int>(timeout));
