diff -Naur a/meson.build b/meson.build
--- a/meson.build	2018-10-19 18:36:42.000000000 +0200
+++ b/meson.build	2018-10-28 02:18:04.031050997 +0200
@@ -381,7 +381,7 @@
 endif
 if with_dri or with_gallium
   if with_glx == 'disabled' and not with_egl and not with_platform_haiku
-    error('building dri or gallium drivers require at least one window system')
+    #error('building dri or gallium drivers require at least one window system')
   endif
 endif
 
