diff -Naur a/src/compiler/Makefile.glsl.am b/src/compiler/Makefile.glsl.am
--- a/src/compiler/Makefile.glsl.am	2018-10-19 18:36:42.000000000 +0200
+++ b/src/compiler/Makefile.glsl.am	2018-10-27 18:08:10.888492000 +0200
@@ -143,14 +143,14 @@
 	glsl/libglsl.la					\
 	$(top_builddir)/src/libglsl_util.la		\
 	$(top_builddir)/src/util/libmesautil.la		\
-	$(PTHREAD_LIBS)
+	$(PTHREAD_LIBS) $(ZLIB_LIBS)
 
 glsl_compiler_SOURCES = \
 	glsl/main.cpp
 
 glsl_compiler_LDADD = \
 	glsl/libstandalone.la \
-	$(CLOCK_LIB)
+	$(CLOCK_LIB) $(ZLIB_LIBS)
 
 glsl_glsl_test_SOURCES = \
 	glsl/test.cpp \
diff -Naur a/src/gallium/targets/osmesa/Makefile.am b/src/gallium/targets/osmesa/Makefile.am
--- a/src/gallium/targets/osmesa/Makefile.am	2018-10-19 18:36:42.000000000 +0200
+++ b/src/gallium/targets/osmesa/Makefile.am	2018-10-27 21:01:24.806076174 +0200
@@ -35,6 +35,8 @@
 	-I$(top_srcdir)/src/gallium/auxiliary \
 	-DGALLIUM_SOFTPIPE
 
+AM_LDFLAGS = $(LDFLAGS)
+
 lib_LTLIBRARIES = lib@OSMESA_LIB@.la
 
 nodist_EXTRA_lib@OSMESA_LIB@_la_SOURCES = dummy.cpp
@@ -64,7 +66,7 @@
 	$(top_builddir)/src/mapi/glapi/libglapi.la \
 	$(SHARED_GLAPI_LIB) \
 	$(OSMESA_LIB_DEPS) \
-	$(CLOCK_LIB) \
+	$(CLOCK_LIB) $(ZLIB_LIBS) \
 	$(LIBUNWIND_LIBS)
 
 if HAVE_GALLIUM_LLVM
