diff -Naur a/CMakeFilters.cmake b/CMakeFilters.cmake
--- a/CMakeFilters.cmake	2018-09-04 04:54:43.000000000 +0200
+++ b/CMakeFilters.cmake	2019-02-15 12:05:36.521553428 +0100
@@ -45,8 +45,8 @@
       if (NOT ZLIB_FOUND)
         find_package (ZLIB) # Legacy find
         if (ZLIB_FOUND)
-          set (LINK_COMP_LIBS ${LINK_COMP_LIBS} ${ZLIB_LIBRARIES})
-          set (LINK_COMP_SHARED_LIBS ${LINK_COMP_SHARED_LIBS} ${ZLIB_LIBRARIES})
+          set (LINK_COMP_LIBS ${LINK_COMP_LIBS} ZLIB::ZLIB)
+          set (LINK_COMP_SHARED_LIBS ${LINK_COMP_SHARED_LIBS} ZLIB::ZLIB)
         endif ()
       endif ()
     endif ()
diff -Naur a/config/cmake/hdf5-config.cmake.in b/config/cmake/hdf5-config.cmake.in
--- a/config/cmake/hdf5-config.cmake.in	2018-09-04 04:54:44.000000000 +0200
+++ b/config/cmake/hdf5-config.cmake.in	2019-02-15 12:07:41.607305564 +0100
@@ -14,6 +14,7 @@
 #-----------------------------------------------------------------------------
 @PACKAGE_INIT@
 
+include(CMakeFindDependencyMacro)
 string(TOUPPER @HDF5_PACKAGE@ HDF5_PACKAGE_NAME)
 
 set (${HDF5_PACKAGE_NAME}_VALID_COMPONENTS
@@ -68,6 +69,10 @@
   set (${HDF5_PACKAGE_NAME}_JAVA_LIBRARIES "${${HDF5_PACKAGE_NAME}_JAVA_LIBRARY}")
 endif ()
 
+if (${HDF5_PACKAGE_NAME}_ENABLE_Z_LIB_SUPPORT)
+    find_dependency(ZLIB)
+endif ()
+
 #-----------------------------------------------------------------------------
 # Directories
 #-----------------------------------------------------------------------------
@@ -105,13 +110,10 @@
 # project which has already built hdf5 as a subproject
 #-----------------------------------------------------------------------------
 if (NOT TARGET "@HDF5_PACKAGE@")
-  if (${HDF5_PACKAGE_NAME}_ENABLE_Z_LIB_SUPPORT AND ${HDF5_PACKAGE_NAME}_PACKAGE_EXTLIBS AND NOT TARGET "zlib")
-    include (@PACKAGE_SHARE_INSTALL_DIR@/@ZLIB_PACKAGE_NAME@/@ZLIB_PACKAGE_NAME@@HDF_PACKAGE_EXT@-targets.cmake)
-  endif ()
   if (${HDF5_PACKAGE_NAME}_ENABLE_SZIP_SUPPORT AND ${HDF5_PACKAGE_NAME}_PACKAGE_EXTLIBS AND NOT TARGET "szip")
     include (@PACKAGE_SHARE_INSTALL_DIR@/@SZIP_PACKAGE_NAME@/@SZIP_PACKAGE_NAME@@HDF_PACKAGE_EXT@-targets.cmake)
   endif ()
-  include (@PACKAGE_SHARE_INSTALL_DIR@/@HDF5_PACKAGE@/@HDF5_PACKAGE@@HDF_PACKAGE_EXT@-targets.cmake)
+  include (@PACKAGE_SHARE_INSTALL_DIR@/@HDF5_PACKAGE@@HDF_PACKAGE_EXT@-targets.cmake)
 endif ()
 
 # Handle default component(static) :
