--- spdlog-0.11.0/include/spdlog/details/os.h	2016-09-18 01:28:41.000000000 +0200
+++ spdlog-0.11.0-mod/include/spdlog/details/os.h	2016-10-15 11:48:20.000000000 +0200
@@ -342,16 +342,11 @@
     else
         return "Unkown error";
 
-#elif defined(__FreeBSD__) || defined(__APPLE__) || defined(ANDROID) || \
-      ((_POSIX_C_SOURCE >= 200112L) && ! _GNU_SOURCE) // posix version
-
+#else
     if (strerror_r(err_num, buf, buf_size) == 0)
         return std::string(buf);
     else
         return "Unkown error";
-
-#else  // gnu version (might not use the given buf, so its retval pointer must be used)
-    return std::string(strerror_r(err_num, buf, buf_size));
 #endif
 }
 
