diff -Naur a/tools/grib_options.c b/tools/grib_options.c
--- a/tools/grib_options.c	2018-04-18 12:36:31.000000000 +0200
+++ b/tools/grib_options.c	2018-07-07 21:59:24.000000000 +0200
@@ -227,7 +227,7 @@
     if (grib_options_on("X:"))
         options->infile_offset=atol(grib_options_get_option("X:"));
 
-#ifndef ECCODES_ON_WINDOWS
+#if !defined ECCODES_ON_WINDOWS && !defined __MINGW32__
     /* Check at compile time to ensure our file offset is at least 64 bits */
     COMPILE_TIME_ASSERT( sizeof(options->infile_offset) >= 8 );
 #endif
diff -Naur a/tools/grib_tools.c b/tools/grib_tools.c
--- a/tools/grib_tools.c	2018-04-18 12:36:31.000000000 +0200
+++ b/tools/grib_tools.c	2018-07-07 22:05:12.000000000 +0200
@@ -295,7 +295,7 @@
             exit(1);
         }
         if (options->infile_offset) {
-#ifndef ECCODES_ON_WINDOWS
+#if !defined ECCODES_ON_WINDOWS && !defined __MINGW32__
             /* Check at compile time to ensure our file offset is at least 64 bits */
             COMPILE_TIME_ASSERT( sizeof(options->infile_offset) >= 8 );
 #endif
