diff -Naur a/deps/agg/src/agg_pixfmt_rgba.cpp b/deps/agg/src/agg_pixfmt_rgba.cpp
--- a/deps/agg/src/agg_pixfmt_rgba.cpp	2018-10-08 11:14:49.000000000 +0200
+++ b/deps/agg/src/agg_pixfmt_rgba.cpp	2019-01-21 19:04:23.000000000 +0100
@@ -3,7 +3,12 @@
 
 #pragma GCC diagnostic push
 #include <mapnik/warning_ignore.hpp>
+#include <boost/version.hpp>
+#if BOOST_VERSION >= 106900
+#include <boost/gil.hpp>
+#else
 #include <boost/gil/gil_all.hpp>
+#endif
 #include <boost/gil/extension/toolbox/hsv.hpp>
 #include <boost/gil/extension/toolbox/hsl.hpp>
 #pragma GCC diagnostic pop
diff -Naur a/deps/boost/gil/extension/toolbox/hsl.hpp b/deps/boost/gil/extension/toolbox/hsl.hpp
--- a/deps/boost/gil/extension/toolbox/hsl.hpp	2018-10-08 11:14:49.000000000 +0200
+++ b/deps/boost/gil/extension/toolbox/hsl.hpp	2019-01-21 19:05:20.000000000 +0100
@@ -16,7 +16,12 @@
 
 #pragma GCC diagnostic push
 #include <mapnik/warning_ignore.hpp>
+#include <boost/version.hpp>
+#if BOOST_VERSION >= 106900
+#include <boost/gil.hpp>
+#else
 #include <boost/gil/gil_all.hpp>
+#endif
 #pragma GCC diagnostic pop
 
 namespace boost { namespace gil {
diff -Naur a/deps/boost/gil/extension/toolbox/hsv.hpp b/deps/boost/gil/extension/toolbox/hsv.hpp
--- a/deps/boost/gil/extension/toolbox/hsv.hpp	2018-10-08 11:14:49.000000000 +0200
+++ b/deps/boost/gil/extension/toolbox/hsv.hpp	2019-01-21 19:05:52.000000000 +0100
@@ -16,7 +16,12 @@
 
 #pragma GCC diagnostic push
 #include <mapnik/warning_ignore.hpp>
+#include <boost/version.hpp>
+#if BOOST_VERSION >= 106900
+#include <boost/gil.hpp>
+#else
 #include <boost/gil/gil_all.hpp>
+#endif
 #pragma GCC diagnostic pop
 
 namespace boost { namespace gil {
diff -Naur a/include/mapnik/image_filter.hpp b/include/mapnik/image_filter.hpp
--- a/include/mapnik/image_filter.hpp	2018-10-08 11:14:49.000000000 +0200
+++ b/include/mapnik/image_filter.hpp	2019-01-21 19:06:19.000000000 +0100
@@ -31,7 +31,12 @@
 
 #pragma GCC diagnostic push
 #include <mapnik/warning_ignore.hpp>
+#include <boost/version.hpp>
+#if BOOST_VERSION >= 106900
+#include <boost/gil.hpp>
+#else
 #include <boost/gil/gil_all.hpp>
+#endif
 #pragma GCC diagnostic pop
 
 #pragma GCC diagnostic push
