diff -Naur a/SConstruct b/SConstruct
--- a/SConstruct	2018-04-12 14:46:14.000000000 +0200
+++ b/SConstruct	2018-07-16 13:46:33.968649381 +0200
@@ -1650,7 +1650,7 @@
                 if plugin == 'gdal':
                     if conf.parse_config('GDAL_CONFIG',checks='--libs'):
                         conf.parse_config('GDAL_CONFIG',checks='--cflags')
-                        libname = conf.get_pkg_lib('GDAL_CONFIG','gdal')
+                        libname = 'gdal'
                         if libname:
                             if not conf.CheckLibWithHeader(libname, details['inc'], details['lang']):
                                 env['SKIPPED_DEPS'].append('gdal')
