diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2025-10-09 15:54:12.778162000 +0200
+++ b/CMakeLists.txt	2025-10-09 15:59:43.677787000 +0200
@@ -120,8 +120,7 @@
 if (CMAKE_CXX_COMPILER_ID STREQUAL "IntelLLVM")
   # Intel CXX compiler based on clang defaults to -ffast-math, which
   # breaks std::isinf(), std::isnan(), etc.
-  set(CMAKE_C_FLAGS ${CMAKE_C_FLAGS} -fno-fast-math)
-  set(CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS} -fno-fast-math)
+  add_compile_options("-fno-fast-math")
 endif ()
 
 set(PROJ_C_WARN_FLAGS "${PROJ_C_WARN_FLAGS}"
