diff --git a/CMakeLists.txt b/CMakeLists.txt
index a0b1fec..6abf28b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,7 +5,7 @@ set(CMAKE_CONFIGURATION_TYPES Debug Release RelWithDebInfo Sanitize)
 
 # Compiler configuration
 set(CMAKE_CXX_EXTENSIONS OFF)
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_C_EXTENSIONS OFF)
 set(CMAKE_C_STANDARD 99)
diff --git a/include/mbgl/util/optional.hpp b/include/mbgl/util/optional.hpp
index abec02d..f987b74 100644
--- a/include/mbgl/util/optional.hpp
+++ b/include/mbgl/util/optional.hpp
@@ -1,13 +1,13 @@
 #pragma once
 
-#include <experimental/optional>
+#include <optional>
 
 namespace mbgl {
 
 template <typename T>
-using optional = std::experimental::optional<T>;
+using optional = std::optional<T>;
 
-using nullopt_t = std::experimental::nullopt_t;
-constexpr nullopt_t nullopt = std::experimental::nullopt;
+using nullopt_t = std::nullopt_t;
+constexpr nullopt_t nullopt = std::nullopt;
 
 } // namespace mbgl
diff --git a/platform/qt/qt.cmake b/platform/qt/qt.cmake
index e651b8c..4b5e3cc 100644
--- a/platform/qt/qt.cmake
+++ b/platform/qt/qt.cmake
@@ -59,7 +59,7 @@ set(MBGL_QT_FILESOURCE_FILES
 )
 
 # Shared library
-add_library(qmapboxgl SHARED
+add_library(qmapboxgl
     platform/qt/include/qmapbox.hpp
     platform/qt/include/qmapboxgl.hpp
     platform/qt/src/qt_conversion.hpp
diff --git a/platform/qt/src/sqlite3.cpp b/platform/qt/src/sqlite3.cpp
index 75befc3..adad5b1 100644
--- a/platform/qt/src/sqlite3.cpp
+++ b/platform/qt/src/sqlite3.cpp
@@ -81,7 +81,7 @@ public:
 };
 
 template <typename T>
-using optional = std::experimental::optional<T>;
+using optional = std::optional<T>;
 
 mapbox::util::variant<Database, Exception> Database::tryOpen(const std::string &filename, int flags) {
     if (!QSqlDatabase::drivers().contains("QSQLITE")) {
diff --git a/src/core-files.json b/src/core-files.json
index 51e48ea..e95adff 100644
--- a/src/core-files.json
+++ b/src/core-files.json
@@ -304,7 +304,8 @@
         "src/mbgl/util/url.cpp",
         "src/mbgl/util/version.cpp",
         "src/mbgl/util/work_request.cpp",
-        "src/parsedate/parsedate.cpp"
+        "src/parsedate/parsedate.cpp",
+        "platform/darwin/src/nsthread.mm"
     ],
     "public_headers": {
         "mbgl/actor/actor.hpp": "include/mbgl/actor/actor.hpp",
diff --git a/src/csscolorparser/csscolorparser.hpp b/src/csscolorparser/csscolorparser.hpp
index 0e16ec4..14c2599 100644
--- a/src/csscolorparser/csscolorparser.hpp
+++ b/src/csscolorparser/csscolorparser.hpp
@@ -27,12 +27,12 @@
 
 #include <string>
 #include <cmath>
-#include <experimental/optional>
+#include <optional>
 
 namespace CSSColorParser {
 
 template <class T>
-using optional = std::experimental::optional<T>;
+using optional = std::optional<T>;
 
 struct Color {
     inline Color() {
diff --git a/src/mbgl/gl/attribute.hpp b/src/mbgl/gl/attribute.hpp
index 8cdb03b..66a8615 100644
--- a/src/mbgl/gl/attribute.hpp
+++ b/src/mbgl/gl/attribute.hpp
@@ -41,6 +41,11 @@ public:
         return std::tie(lhs.attributeType, lhs.attributeSize, lhs.attributeOffset, lhs.vertexBuffer, lhs.vertexSize, lhs.vertexOffset)
             == std::tie(rhs.attributeType, rhs.attributeSize, rhs.attributeOffset, rhs.vertexBuffer, rhs.vertexSize, rhs.vertexOffset);
     }
+
+    friend bool operator!=(const AttributeBinding& lhs,
+                           const AttributeBinding& rhs) {
+        return !(lhs == rhs);
+    }
 };
 
 using AttributeBindingArray = std::vector<optional<AttributeBinding>>;
diff --git a/src/mbgl/text/glyph_manager.cpp b/src/mbgl/text/glyph_manager.cpp
index b947ef7..aa50bf7 100644
--- a/src/mbgl/text/glyph_manager.cpp
+++ b/src/mbgl/text/glyph_manager.cpp
@@ -138,7 +138,7 @@ void GlyphManager::notify(GlyphRequestor& requestor, const GlyphDependencies& gl
             if (it != entry.glyphs.end()) {
                 glyphs.emplace(*it);
             } else {
-                glyphs.emplace(glyphID, std::experimental::nullopt);
+                glyphs.emplace(glyphID, std::nullopt);
             }
         }
     }
diff --git a/platform/default/src/mbgl/storage/sqlite3.cpp b/platform/default/src/mbgl/storage/sqlite3.cpp
index 0017dc4..806b85f 100644
--- a/platform/default/src/mbgl/storage/sqlite3.cpp
+++ b/platform/default/src/mbgl/storage/sqlite3.cpp
@@ -6,7 +6,7 @@
 #include <cstring>
 #include <cstdio>
 #include <chrono>
-#include <experimental/optional>
+#include <optional>
 
 #include <mbgl/util/traits.hpp>
 #include <mbgl/util/logging.hpp>
@@ -99,7 +99,7 @@ public:
 };
 
 template <typename T>
-using optional = std::experimental::optional<T>;
+using optional = std::optional<T>;
 
 
 #ifndef NDEBUG
