diff -Naur a/configure b/configure
--- a/configure	2018-06-23 17:05:47.000000000 +0200
+++ b/configure	2018-07-08 12:23:52.000000000 +0200
@@ -20545,20 +20545,10 @@
 
 
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for 64bit integer type" >&5
-$as_echo_n "checking for 64bit integer type... " >&6; }
-
-  echo 'int main() { return (int)(long long)(0); }' >> conftest.c
-  if test -z "`${CC} ${CFLAGS} -o conftest conftest.c 2>&1`" ; then
-
+$as_echo_n "checking for 64bit integer type... " >&6;
 $as_echo "#define HAVE_LONG_LONG 1" >>confdefs.h
-
-    { $as_echo "$as_me:${as_lineno-$LINENO}: result: long long" >&5
 $as_echo "long long" >&6; }
-  else
-    as_fn_error $? "long long not found" "$LINENO" 5
-  fi
-  rm -f conftest*
-
+  
 
 
 # Check whether --with-unix-stdio-64 was given.
