diff -Naur a/apps/makefile.vc b/apps/makefile.vc
--- a/apps/makefile.vc	2018-05-04 18:05:45.000000000 +0200
+++ b/apps/makefile.vc	2018-06-21 09:42:50.520400000 +0200
@@ -20,13 +20,19 @@
 !ENDIF
 
 
-default:	gdal_translate.exe gdalinfo.exe gdalserver.exe gdaladdo.exe gdalwarp.exe \
-		nearblack.exe gdalmanage.exe gdalenhance.exe gdaltransform.exe\
-		gdaldem.exe gdallocationinfo.exe gdalsrsinfo.exe $(OGR_PROGRAMS) $(GNM_PROGRAMS)
-
-all:	default multireadtest.exe \
-			dumpoverviews.exe gdalwarpsimple.exe gdalflattenmask.exe \
-			gdaltorture.exe gdal2ogr.exe test_ogrsf.exe
+!IFDEF BUILD_TOOLS
+default:    gdal_translate.exe gdalinfo.exe gdalserver.exe gdaladdo.exe gdalwarp.exe \
+        nearblack.exe gdalmanage.exe gdalenhance.exe gdaltransform.exe\
+        gdaldem.exe gdallocationinfo.exe gdalsrsinfo.exe $(OGR_PROGRAMS) $(GNM_PROGRAMS)
+
+all:    default multireadtest.exe \
+            dumpoverviews.exe gdalwarpsimple.exe gdalflattenmask.exe \
+            gdaltorture.exe gdal2ogr.exe test_ogrsf.exe
+
+!ELSE
+default: ;
+all: default
+!ENDIF
 OBJ = commonutils.obj gdalinfo_lib.obj gdal_translate_lib.obj gdalwarp_lib.obj ogr2ogr_lib.obj \
 	gdaldem_lib.obj nearblack_lib.obj gdal_grid_lib.obj gdal_rasterize_lib.obj gdalbuildvrt_lib.obj
 
@@ -207,4 +213,6 @@
 	-del *.exp
 
 install:	default
+!IFDEF BUILD_TOOLS
 	copy *.exe $(BINDIR)
+!ENDIF
\ No newline at end of file
