diff -Naur a/apps/GNUmakefile b/apps/GNUmakefile
--- a/apps/GNUmakefile	2018-06-23 17:05:46.000000000 +0200
+++ b/apps/GNUmakefile	2018-07-01 13:12:44.000000000 +0200
@@ -41,7 +41,7 @@
 	gdaltorture$(EXE) gdal2ogr$(EXE) test_ogrsf$(EXE) \
 	gdalasyncread$(EXE) testreprojmulti$(EXE)
 
-default:	gdal-config-inst gdal-config $(BIN_LIST)
+default:	gdal-config-inst gdal-config
 
 all: default $(NON_DEFAULT_LIST)
 
@@ -215,7 +215,6 @@
 
 
 install: default
-	for f in $(BIN_LIST) ; do $(INSTALL) $$f $(DESTDIR)$(INST_BIN) ; done
 	$(INSTALL_DATA) gdal_utils.h $(DESTDIR)$(INST_INCLUDE)
 	$(INSTALL) gdal-config-inst $(DESTDIR)$(INST_BIN)/gdal-config
 
