diff -Naur a/cmake/ecbuild_get_test_data.cmake b/cmake/ecbuild_get_test_data.cmake
--- a/cmake/ecbuild_get_test_data.cmake	2018-04-18 12:36:27.000000000 +0200
+++ b/cmake/ecbuild_get_test_data.cmake	2018-07-07 21:35:24.000000000 +0200
@@ -407,8 +407,8 @@
 
         # The option /fast disables dependency checking on a target, see
         # https://cmake.org/Wiki/CMake_FAQ#Is_there_a_way_to_skip_checking_of_dependent_libraries_when_compiling.3F
-        if( WIN32 )
-          set( _fast "\fast" )
+        if( WIN32 AND NOT MINGW)
+          set( _fast "\\fast" )
         else()
           set( _fast "/fast" )
         endif()
