diff -Naru a/netCDFConfig.cmake.in b/netCDFConfig.cmake.in
--- a/netCDFConfig.cmake.in	2021-08-18 19:49:05.000000000 +0200
+++ b/netCDFConfig.cmake.in	2023-03-21 19:54:08.203081700 +0100
@@ -12,6 +12,14 @@
 set(netCDF_LIBRARIES netCDF::netcdf)
 
 # include target information
+include(CMakeFindDependencyMacro)
+find_dependency(HDF5 CONFIG)
+if (@ENABLE_DAP@)
+    find_dependency(CURL CONFIG)
+endif ()
+if(@ENABLE_NCZARR_ZIP@)
+    find_dependency(libzip CONFIG)
+endif()
 include("${CMAKE_CURRENT_LIST_DIR}/netCDFTargets.cmake")
 
 # Compiling Options
