diff -Naur a/liblib/CMakeLists.txt b/liblib/CMakeLists.txt
--- a/liblib/CMakeLists.txt	2016-11-21 19:27:08.000000000 +0100
+++ b/liblib/CMakeLists.txt	2019-02-15 13:38:49.484631371 +0100
@@ -35,7 +35,7 @@
 
 SET(TLL_LIBS "")
 
-SET(TLL_LIBS ${TLL_LIBS} ${HAVE_LIBM} ${ZLIB_LIBRARY})
+SET(TLL_LIBS ${TLL_LIBS} ${HAVE_LIBM} ZLIB::ZLIB)
 
 # Add extra dependencies specified via NC_EXTRA_DEPS
 SET(TLL_LIBS ${TLL_LIBS} ${EXTRA_DEPS})
@@ -45,23 +45,7 @@
 ENDIF()
 
 IF(USE_HDF5 OR USE_NETCDF4)
-  IF(NOT MSVC)
-    # Some version of cmake define HDF5_hdf5_LIBRARY instead of
-    # HDF5_LIBRARY. Same with HDF5_HL_LIBRARIES
-    IF(HDF5_hdf5_LIBRARY AND NOT HDF5_C_LIBRARIES)
-      SET(HDF5_C_LIBRARIES ${HDF5_hdf5_LIBRARY})
-    ENDIF()
-    IF(HDF5_hdf5_hl_LIBRARY AND NOT HDF5_HL_LIBRARIES)
-      SET(HDF5_HL_LIBRARIES ${HDF5_hdf5_hl_LIBRARY})
-    ENDIF()
-    # The order of the libraries is important here for static
-    # builds:
-    # Make sure that HDF5_C_LIBRARY appears *after*
-    # HDF5_HL_LIBRARY.
-    SET(TLL_LIBS ${HDF5_HL_LIBRARIES} ${HDF5_C_LIBRARIES} ${TLL_LIBS} ${SZIP_LIBRARY})
-  ELSE() # Windows CMake defines HDF5_LIBRARIES.
-    SET(TLL_LIBS ${HDF5_LIBRARIES} ${TLL_LIBS} ${SZIP_LIBRARY})
-  ENDIF()
+  SET(TLL_LIBS ${TLL_LIBS} hdf5::hdf5_hl-static hdf5::hdf5-static ${SZIP_LIBRARY})
 ENDIF()
 
 IF(USE_DAP)
