diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2019-02-16 08:11:14.000000000 +0100
+++ b/CMakeLists.txt	2019-02-16 13:29:46.000000000 +0100
@@ -1852,7 +1852,7 @@
 STRING(REPLACE "-lhdf5::hdf5_hl-shared" "-lhdf5_hl" NC_LIBS ${NC_LIBS})
 STRING(REPLACE "-lhdf5::hdf5-static" "-lhdf5" NC_LIBS ${NC_LIBS})
 STRING(REPLACE "-lhdf5::hdf5_hl-static" "-lhdf5_hl" NC_LIBS ${NC_LIBS})
-
+STRING(REPLACE "-lZLIB::ZLIB" "-lz" NC_LIBS ${NC_LIBS})
 
 STRING(REPLACE ";" " " LINKFLAGS "${LINKFLAGS}")
 
