diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2019-02-16 08:11:14.000000000 +0100
+++ b/CMakeLists.txt	2019-02-16 13:15:20.000000000 +0100
@@ -832,15 +832,6 @@
 ENDIF()
 
 
-# Check for the math library so it can be explicitly linked.
-IF(NOT WIN32)
-  FIND_LIBRARY(HAVE_LIBM NAMES math m libm)
-  MESSAGE(STATUS "Found Math library: ${HAVE_LIBM}")
-  IF(NOT HAVE_LIBM)
-    MESSAGE(FATAL_ERROR "Unable to find the math library.")
-  ENDIF()
-ENDIF()
-
 # Option to Enable DAP long tests, remote tests.
 OPTION(ENABLE_DAP_LONG_TESTS "Enable DAP long tests." OFF)
 OPTION(ENABLE_DAP_REMOTE_TESTS "Enable DAP remote tests." ON)
